packages/zod-mock/src/lib/zod-mock.ts

```diff
@@ -369,10 +375,9 @@ function parseUnion(
 ) {
   const fakerInstance = options?.faker || faker;
   // Map the options to various possible mock values
-  const mockOptions = zodRef._def.options.map((option) =>
-    generateMock(option, options)
-  );
-  return fakerInstance.helpers.arrayElement(mockOptions);
+  const potentialCases = [...zodRef._def.options.values()];
+  const mocked = fakerInstance.helpers.arrayElement(potentialCases);
+  return generateMock(mocked, options);
```

Comment:
This is more efficient as it only generates a single value, rather than one for every element.